docs/reference/changelog.rst

```diff
@@ -0,0 +1,264 @@
+Changelog
+*********
+
+..
+  release template:
+
+  X.Y.Z (YYYY-MM-DD)
+  ------------------
+
+  Core
+  ====
+
+  # for everything related to the lifecycle of packing a charm
+
+  Bases
+  #####
+
+  <distro>@<series>
+  """""""""""""""""
+  (order from newest base to oldest base)
+
+  Plugins
+  #######
+
+  <plugin>
+  """"""""
+
+  List plugins
+  """"""""""""
+
+  Extensions
+  ##########
+
+  <extension>
+  """""""""""
+
+  Expand extensions
+  """""""""""""""""
+
+  List extensions
+  """""""""""""""
+
+  Metadata
+  ########
+
+  Sources
+  #######
+
+  Components
+  ##########
+
+  Command line
+  ============
+
+  # for command line and UX changes
+
+  Linter
+  ======
+
+  Init
+  ====
+
+  Metrics
+  =======
+
+  Names
+  =====
+
+  Remote build
+  ============
+
+  Store
+  =====
+
+  Documentation
+  =============
+
+  For a complete list of commits, see the `X.Y.Z`_ release on GitHub.
+
+
+3.1.2 (2024-08-07)
+------------------
+
+Bug fixes:
```

Comment:
Is Bug fixes here and in 3.1.1 in a "plain" style on purpose? It looks very different from 3.1.2, where Bugfixes (different spelling) is an actual section:

![image](https://github.com/user-attachments/assets/7a479a54-47ca-42d5-adfa-077a06fef63e)

Reply:
Not intentional - I just copied the notes from the various releases.

I've removed the bug fixes headers from patch releases (it's redundant since patch releases should just be bug fixes) and made the 3.1.0 one two words.

Reply:
Yes, at minimum the headings should be consistent throughout the document, even if it means they no longer mirror the original text.